### PR TITLE
fix(store): let setup --no-init-git run inside a git repository

### LINKED ISSUE #1883: `store setup --no-init-git` is refused when `$HOME` is a Git repository
## Description

`openspec store setup` refuses a `--path` inside another Git repository
(`store_setup_inside_git_repo`). Its own comment gives the reason:

> Stores may be Git-backed, but creating one inside an implementation repo is
> almost always an accidental nested-repo setup.

The check also runs with `--no-init-git`, which creates no repository at all, so
there is nothing to nest. The core already has a bypass,
`allowInsideGitRepository`, but no CLI code ever sets it.

Many people keep their dotfiles as a Git repository at `$HOME`. For them, the
documented onboarding layout (`~/openspec/<id>`) cannot be set up with any flag.

## Reproduction

```bash
git init ~                       # a dotfiles repository
openspec store setup team-plans --path ~/openspec/team-plans --no-init-git --json
```

## Expected

The store is created and registered, with no `.git` of its own.

## Actual

```
[control: HOME not a repo]                  rc=0 "registered":true
[HOME is a dotfiles repo, --no-init-git]    rc=1 "code":"store_setup_inside_git_repo"
store folder created? no
```

The only workaround, creating the folder by hand and then running
`store register`, is not documented.

## Root cause

- The guard runs in `prepareSetupPlan` before any Git decision is known
  (`src/core/store/operations.ts:483-487`).
- The bypass option `allowInsideGitRepository`
  (`src/core/store/operations.ts:137`, `:317-319`) is never passed by the CLI.
  `prepareSetupInput` (`src/commands/store.ts:295-300`) receives the command
  options, including `initGit`, and ignores them (`_options`).

## Environment

- OpenSpec 1.13.0 (reproduced on the published npm package and on a build of
  `main` @ `9d4e597`)
- Node 22, Linux
- Every run used its own temporary `HOME`, `XDG_CONFIG_HOME` and `XDG_DATA_HOME`

### PR DESCRIPTION
**Pull request by dwin-gharibi**
Closes #1883.

## Why

`openspec store setup` refuses a `--path` inside another Git repository
(`store_setup_inside_git_repo`). Its comment gives the reason: "creating one
inside an implementation repo is almost always an accidental nested-repo
setup". The check also ran with `--no-init-git`, which creates no repository, so
there is nothing to nest.

Users who keep their home directory as a dotfiles repository could not set up a
store at the recommended `~/openspec/<id>` with any flag. The core already had a
bypass, `allowInsideGitRepository`, but nothing passed it. The CLI's
`prepareSetupInput` received the command options, including `initGit`, and
ignored them (`_options`).

## What Changes

- `prepareSetupPlan` skips the nested-repo check when `initGit === false`. The
  decision lives in the core, next to the check and its comment, so the
  programmatic `setupStore` behaves the same as the CLI.
- The CLI forwards an explicit `--init-git` / `--no-init-git` into the prepare
  step.
- `docs-lab/reference/cli.md` explains the check and when `--no-init-git` skips it.

Unchanged:

- the default setup and an explicit `--init-git` still refuse a path inside
  another repository
- the store never records the enclosing repository's `origin`. The existing
  at-root check in `resolveBackendWithObservedOrigin` already ensured this; a
  test now covers it.

### Existing tests changed

Two tests in `test/commands/store.test.ts` passed `--no-init-git` while
asserting the refusal, so they encoded this bug:

- `rejects explicit setup paths inside an existing Git repo in non-interactive mode`
- `rejects setup paths inside git-like parents when git cannot resolve the repo`

They now run the default setup, which initializes Git and is exactly what the
check is for. They still assert the refusal and that nothing was created. The
check runs in prepare, before the Git identity preflight, so they still need no
Git identity.

## Testing

`test/commands/store-setup-no-init-git.test.ts`: 7 tests, run through the built
CLI. Written first and watched fail:

- **Before the fix (tests only, on `9d4e597`):** 4 failed, 3 passed. Every
  `--no-init-git` setup under a `$HOME` repository exited 1, refused by the
  nested-repo guard (`expected 1 to be +0`). The not-a-repository control and
  the two still-refused cases (default and `--init-git`) passed.
- **After the fix:** 7 passed.

`test/commands/store.test.ts`, holding the two updated cases, passes all 43
tests both before and after the fix. Before the fix those two cases still assert
the refusal, which the default setup still gives.

### CI parity

Run in an isolated sandbox on `9d4e597` under Node 20.19.0 and pnpm 10.34.5,
with the steps from `.github/workflows/ci.yml`:

| Step | Result |
|---|---|
| `pnpm run build` | pass |
| `pnpm exec tsc --noEmit` | pass |
| `pnpm lint` | pass |
| `pnpm test` (`VITEST_MAX_WORKERS=4`) | 4561 passed, 9 failed (4570) |
| `pnpm test` on clean `9d4e597`, same sandbox | 4555 passed, 8 failed (4563) |

The full-suite failures are all timeouts — `Test timed out in 10000ms`, or
`spawnSync npm ETIMEDOUT` in the npm packing test — in CLI-heavy files:
`store-references`, `store-remote`, `store-root-selection`, `workset` and
`package-install-scripts`. Two checks say the sandbox causes them, not this
change:

- clean `9d4e597` fails the same files the same way on the same machine
- every one of those files passes alone (`--maxWorkers=2`), on this branch and
  on clean `9d4e597`: `store-references` 8/8, `store-remote` 19/19,
  `store-root-selection` 44/44, `store.test.ts` 43/43, `workset` 41/41,
  `package-install-scripts` 8/8

The branch adds exactly its 7 new tests: 4570 = 4563 + 7.

Edge cases covered:

- control: `--no-init-git` outside any repository
- `$HOME` is a repository:
  - setup succeeds and registers the store
  - `git` reports `is_repository/initialized/committed: false`
  - metadata and config are written
- no `.git` is created in the store, and the enclosing repository gets no commit
- an `origin` on the enclosing repository is never recorded as the store's remote
- a rerun with `--no-init-git` is an already-registered no-op
- the default setup is still refused, and nothing is created
- an explicit `--init-git` is still refused

## Changeset

`.changeset/store-setup-no-init-git-inside-repo.md` (patch).

## Notes for review

- A rerun without `--no-init-git` of a store already registered inside a
  repository is still refused, because the check runs before the registry
  lookup that makes reruns Git-free. Passing `--no-init-git` again works. I kept
  the change to the explicit flag rather than reordering the prepare step.
- `fix-store-remove-nested` also edits `src/core/store/operations.ts`, but in
  `removeStore`. The two should merge without conflicts.

## Summary by CodeRabbit

- **New Features**
  - `openspec store setup --no-init-git` now supports creating stores inside existing Git repositories without initializing nested repositories or recording the enclosing repository’s remote.
  - Re-running setup in this mode reports the store as already registered without making changes.

- **Documentation**
  - Clarified Git repository restrictions and when to use `--no-init-git`.

- **Bug Fixes**
  - Preserved the existing rejection behavior for default setup and explicit `--init-git` inside Git repositories.

### CODE REVIEW

**Comment by clay-good** [comment]
Hardening pass, two commits on top of the fix:

- e0b0b0ad: moved the nested-repo / `--no-init-git` note from legacy `docs/cli.md` to the canonical `docs-lab/reference/cli.md` (store setup section). docs-lab changes need @TabishB review.
- 53a800d6: canonicalized the path identity assertions in `store-setup-no-init-git.test.ts` (CodeRabbit thread).

Verified: the issue repro fails on main (`store_setup_inside_git_repo`, rc=1) and passes on this branch; the new test file fails 4/7 with the src change reverted and passes 7/7 with it; `store.test.ts` 43/43, tsc and lint clean.

File: test/commands/store-setup-no-init-git.test.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+  it('sets up and registers the store when $HOME is a Git repository', async () => {
+    makeHomeADotfilesRepo();
+
+    const result = await setup(['--no-init-git']);
+
+    expect(result.exitCode).toBe(0);
+    const payload = parseJson(result);
+    expect(payload.status).toEqual([]);
+    expect(payload.store.root).toBe(storeRoot);
```
[the remark this answers is not in the document]
Comment (clay-good) [reply test/commands/store-setup-no-init-git.test.ts]: Fixed in 53a800d6: `payload.store.root` and the registry `local_path` are both canonicalized with `fs.realpathSync.native()` before comparing with `storeRoot`.

**Review by alfred-openspec — APPROVED** [review]
The no-init-git path now bypasses only the nested-repository creation guard, still avoids inheriting the enclosing origin, and preserves the default and explicit init-git refusals. Focused store tests pass. @TabishB remains requested for final docs review.
